lib/oli/activities/reports/providers/OliLikert.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Oli.Activities.Reports.Providers.OliLikert do
+  @behaviour Oli.Activities.Reports.Renderer
+
+  alias Oli.Delivery.Attempts.Core
+
+  @impl Oli.Activities.Reports.Renderer
+  def render(
+        %Oli.Rendering.Context{enrollment: enrollment, user: user} = context,
+        %{"activityId" => activity_id} = _element
+      ) do
+    with %{choices: choices, data: data} <-
+           process_attempt_data(enrollment.section_id, user.id, activity_id) do
+      data_url = "/api/v1/activity/report/#{enrollment.section_id}/#{activity_id}"
+
+      prompts = [~s|<div><ul>|, prompts_from_items(data), "</ul></div>"]
+
+      groups =
+        data
+        |> Enum.reduce([], fn a, c ->
+          case Map.get(a, :group) do
+            nil -> c
+            g -> c ++ [g]
+          end
+        end)
+        |> Enum.uniq()
+        |> Enum.join(" ---- ")
+
+      {:ok, first} =
+        choices
+        |> List.first()
+        |> JSONPointer.get("/content/0/children/0/text")
+
+      {:ok, last} =
+        choices
+        |> List.last()
+        |> JSONPointer.get("/content/0/children/0/text")
+
+      visuals = [~s|
+    <div class="flex flex-row">
+      <div class="flex flex-col ml-2">
+        <div>#{last}</div>
+        <div class="grow"></div>
+        <div class="mb-8">#{first}</div>
+      </div>
+      <div class="grow">|, visualization(context, data_url, groups), ~s|</div>
+    </div>|]
+
+      {:ok, [visuals, prompts]}
+    else
+      e -> {:error, e}
+    end
+  end
+
+  @impl Oli.Activities.Reports.Renderer
+  def report_data(section_id, user_id, activity_id) do
+    Map.get(process_attempt_data(section_id, user_id, activity_id), :data)
+  end
+
+  defp process_attempt_data(section_id, user_id, activity_id) do
+    activity_attempt =
+      Core.get_latest_activity_attempt(section_id, user_id, activity_id)
+      |> Oli.Repo.preload(:part_attempts)
+
+    part_attempts_by_id =
+      Enum.reduce(activity_attempt.part_attempts, %{}, fn a, c -> Map.put(c, a.part_id, a) end)
+
+    data =
+      Map.get(activity_attempt.revision.content, "items")
```

Comment:
The `revision` field should not be directly used here, as doing so would not take into account dynamic questions.  Instead, one must select the model, taking into account the `transformed_model` attribute of the activity attempt.  If it is not `nil`, use that.  If it is `nil`, using the content of the `revision`.

Reply:
Issue addessed